src/west/app/config.py

```diff
@@ -135,18 +140,27 @@ def do_run(self, args, user_args):
                 self.parser.error('-l cannot be combined with name argument')
             elif delete:
                 self.parser.error('-l cannot be combined with -d or -D')
+            elif args.append:
+                self.parser.error('-l cannot be combined with -a')
         elif not args.name:
             self.parser.error('missing argument name '
                               '(to list all options and values, use -l)')
-        elif args.delete and args.delete_all:
-            self.parser.error('-d cannot be combined with -D')
+        elif args.delete:
+            if args.delete_all:
+                self.parser.error('-d cannot be combined with -D')
+            elif args.append:
+                self.parser.error('-d cannot be combined with -a')
+        elif args.value is None and args.append:
+            self.parser.error('-a requires a value')
```

Comment:
These checks are becoming a mess, can we use a [mutually exclusive group](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_mutually_exclusive_group) instead?